native/src/App.tsx

```diff
@@ -36,17 +40,36 @@ const linking: LinkingOptions = {
       [REDIRECT_ROUTE]: '*'
     }
   },
-  getStateFromPath: path => {
-    return {
-      index: 0,
-      routes: [
-        {
-          name: REDIRECT_ROUTE,
-          params: {
-            url: `https://${path}`
-          }
+  getStateFromPath: path => ({
+    index: 0,
+    routes: [
+      {
+        name: REDIRECT_ROUTE,
+        params: {
+          url: `https://${path}`
         }
-      ]
+      }
+    ]
+  }),
+  subscribe: (listener: (url: string) => void) => {
+    const onReceiveURL = ({ url }: { url: string }) => listener(url)
+
+    Linking.addEventListener('url', onReceiveURL)
+
+    const unsubscribeNotification = messaging().onNotificationOpenedApp(() => {
+      listener(
+        urlFromRouteInformation({
```

Comment:
Maybe it would be cleaner to support links like: `https://integreat.app///placeholder/news/local`.
Empty string would evaluate also to `false`. Maybe that would be simpler.

Reply:
Tried it, makes everything even worse in the pathname parser as we don't know which are actual path segments. Imo it is also clearer with dedicated placeholders.